client/web/src/repo/blob/Blob.tsx

```diff
@@ -114,78 +104,100 @@ const domFunctions = {
     },
 }
 
-export class Blob extends React.Component<BlobProps, BlobState> {
-    /** Emits with the latest Props on every componentDidUpdate and on componentDidMount */
-    private componentUpdates = new Subject<BlobProps>()
-
-    /** Emits whenever the ref callback for the code element is called */
-    private codeViewElements = new Subject<HTMLElement | null>()
-    private nextCodeViewElement = (element: HTMLElement | null): void => this.codeViewElements.next(element)
-
-    /** Emits whenever the ref callback for the blob element is called */
-    private blobElements = new Subject<HTMLElement | null>()
-    private nextBlobElement = (element: HTMLElement | null): void => this.blobElements.next(element)
-
-    /** Emits whenever the ref callback for the hover element is called */
-    private hoverOverlayElements = new Subject<HTMLElement | null>()
-    private nextOverlayElement = (element: HTMLElement | null): void => this.hoverOverlayElements.next(element)
-
-    /** Emits when the close button was clicked */
-    private closeButtonClicks = new Subject<MouseEvent>()
-    private nextCloseButtonClick = (event: MouseEvent): void => this.closeButtonClicks.next(event)
-
-    /** Subscriptions to be disposed on unmout */
-    private subscriptions = new Subscription()
-
-    constructor(props: BlobProps) {
-        super(props)
-        this.state = {
-            lineDecorationAttachmentIDs: {},
-        }
-
-        /** Emits parsed positions found in the URL */
-        const locationPositions: Observable<LineOrPositionOrRange> = this.componentUpdates.pipe(
-            map(props => parseHash(props.location.hash)),
-            distinctUntilChanged((a, b) => isEqual(a, b)),
-            share()
-        )
+export const Blob: React.FunctionComponent<BlobProps> = props => {
+    const { location, isLightTheme, extensionsController, model } = props
+
+    // Element reference subjects passed to `hoverifier`. They must be `ReplaySubjects` because
+    // the ref callback is called before hoverifier is created in `useEffect`
+    const [blobElements, nextBlobElement] = useSubject<HTMLElement | null>(replaySubjectFactory)
+    const [hoverOverlayElements, nextOverlayElement] = useSubject<HTMLElement | null>(replaySubjectFactory)
+
+    const [codeViewElements, nextCodeViewElement] = useSubject<HTMLElement | null>(replaySubjectFactory)
+    const codeViewReference = React.useRef<HTMLElement | null>()
```

Comment:
Nit: can we cherry-pick the import for consistency?
```suggestion
    const codeViewReference = useRef<HTMLElement | null>()
```